libticalcs/trunk/src/cmdz80.cc

```diff
@@ -540,7 +540,10 @@ int ti82_send_asm_exec(CalcHandle* handle, VarEntry * var)
 	endptr = (var->name[0] == 0x24 ? fpBase : tempMem);
 	buffer[8] = 0x2a; buffer[9] = LSB(endptr); buffer[10] = MSB(endptr);
 	/* ld de, -program_size */
+#pragma warning(push)
```

Comment:
`#pragma` isn't portable in the first place, but would `#pragma msvc ...` be possible at least for the `warning(disable:4146)` ?

Reply:
Well I know, but gcc and clang just warn that it's an unknown pragma and ignore it anyway :P

But contrary to those two compilers with a compiler-name-as-first-arg, it looks like `#pragma msvc` isn't a thing...
So I suppose we could add `#ifdef _MSC_VER` etc. but that's a total of 5 extra lines just to disable a warning, bleh.

Reply:
Taking a step back... fixing the code would be better :)
offset should be `int32_t`, and after adding an early bounds check against < 2 and >= 32768 (both models have 32 KB of physical RAM) on `var->size` below `VALIDATE_VARENTRY(var)`, something like `offset = -(int32_t)((uint16_t)var->size - 2);` should do the job, hopefully without the compilers complaining about numerical value truncation or overflows.

Reply:
Yeah that's probably better. We'll have to compare the actual output in `buffer` just in case...